[Snyk] Security upgrade python from 3.7-slim to 3.14.0rc1-slim

benjaminhuo opened the pull request:
![snyk-top-banner](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/pr-banner-default.svg)

### Snyk has created this PR to fix 3 vulnerabilities in the dockerfile dependencies of this project.

Keeping your Docker base image up-to-date means you’ll benefit from security fixes in the latest version of your chosen image.

#### Snyk changed the following file(s):

- `examples/openfunction_dapr_trigger/caller/Dockerfile`

We recommend upgrading to `python:3.14.0rc1-slim`, as this image has only **44** known vulnerabilities. To do this, merge this pull request, then verify your application still works as expected.



#### Vulnerabilities that will be fixed with an upgrade:

|  | Issue | Score | 
:-------------------------:|:-------------------------|:-------------------------
![high severity](https://res.cloudinary.com/snyk/image/upload/w_20,h_20/v1561977819/icon/h.png 'high severity') | Out-of-bounds Write <br/>[SNYK-DEBIAN12-GLIBC-5927132](https://snyk.io/vuln/SNYK-DEBIAN12-GLIBC-5927132) | &nbsp;&nbsp;**829**&nbsp;&nbsp; 
![high severity](https://res.cloudinary.com/snyk/image/upload/w_20,h_20/v1561977819/icon/h.png 'high severity') | Out-of-bounds Write <br/>[SNYK-DEBIAN12-GLIBC-5927132](https://snyk.io/vuln/SNYK-DEBIAN12-GLIBC-5927132) | &nbsp;&nbsp;**829**&nbsp;&nbsp; 
![high severity](https://res.cloudinary.com/snyk/image/upload/w_20,h_20/v1561977819/icon/h.png 'high severity') | Out-of-bounds Write <br/>[SNYK-DEBIAN12-GLIBC-6210098](https://snyk.io/vuln/SNYK-DEBIAN12-GLIBC-6210098) | &nbsp;&nbsp;**721**&nbsp;&nbsp; 
![high severity](https://res.cloudinary.com/snyk/image/upload/w_20,h_20/v1561977819/icon/h.png 'high severity') | Out-of-bounds Write <br/>[SNYK-DEBIAN12-GLIBC-6210098](https://snyk.io/vuln/SNYK-DEBIAN12-GLIBC-6210098) | &nbsp;&nbsp;**721**&nbsp;&nbsp; 
![high severity](https://res.cloudinary.com/snyk/image/upload/w_20,h_20/v1561977819/icon/h.png 'high severity') | Allocation of Resources Without Limits or Throttling <br/>[SNYK-DEBIAN12-SYSTEMD-6277507](https://snyk.io/vuln/SNYK-DEBIAN12-SYSTEMD-6277507) | &nbsp;&nbsp;**721**&nbsp;&nbsp; 



---

> [!IMPORTANT]
>
> - Check the changes in this PR to ensure they won't cause issues with your project.
> - Max score is 1000. Note that the real score may have changed since the PR was raised.
> - This PR was automatically created by Snyk using the credentials of a real user.

---

**Note:** _You are seeing this because you or someone else with access to this repository has authorized Snyk to open fix PRs._

For more information: <img src="https://api.segment.io/v1/pixel/track?data=eyJ3cml0ZUtleSI6InJyWmxZcEdHY2RyTHZsb0lYd0dUcVg4WkFRTnNCOUEwIiwiYW5vbnltb3VzSWQiOiI4NzgyYjU2Ni03Yzg2LTQ0OTItODhiZS1mYWJjNjM0OGJiMjUiLCJldmVudCI6IlBSIHZpZXdlZCIsInByb3BlcnRpZXMiOnsicHJJZCI6Ijg3ODJiNTY2LTdjODYtNDQ5Mi04OGJlLWZhYmM2MzQ4YmIyNSJ9fQ==" width="0" height="0"/>
🧐 [View latest project report](https://app.snyk.io/org/benjaminhuo/project/6522662d-8795-40dd-afbf-28a2540a1c6b?utm_source&#x3D;github&amp;utm_medium&#x3D;referral&amp;page&#x3D;fix-pr)
📜 [Customise PR templates](https://docs.snyk.io/scan-using-snyk/pull-requests/snyk-fix-pull-or-merge-requests/customize-pr-templates?utm_source=github&utm_content=fix-pr-template)
🛠 [Adjust project settings](https://app.snyk.io/org/benjaminhuo/project/6522662d-8795-40dd-afbf-28a2540a1c6b?utm_source&#x3D;github&amp;utm_medium&#x3D;referral&amp;page&#x3D;fix-pr/settings)
📚 [Read about Snyk's upgrade logic](https://docs.snyk.io/scan-with-snyk/snyk-open-source/manage-vulnerabilities/upgrade-package-versions-to-fix-vulnerabilities?utm_source=github&utm_content=fix-pr-template)

---

**Learn how to fix vulnerabilities with free interactive lessons:**

🦉 [Allocation of Resources Without Limits or Throttling](https://learn.snyk.io/lesson/no-rate-limiting/?loc&#x3D;fix-pr)

[//]: # 'snyk:metadata:{"customTemplate":{"variablesUsed":[],"fieldsUsed":[]},"dependencies":[{"name":"python","from":"3.7-slim","to":"3.14.0rc1-slim"}],"env":"prod","issuesToFix":["SNYK-DEBIAN12-GLIBC-5927132","SNYK-DEBIAN12-GLIBC-5927132","SNYK-DEBIAN12-GLIBC-6210098","SNYK-DEBIAN12-GLIBC-6210098","SNYK-DEBIAN12-SYSTEMD-6277507"],"prId":"8782b566-7c86-4492-88be-fabc6348bb25","prPublicId":"8782b566-7c86-4492-88be-fabc6348bb25","packageManager":"dockerfile","priorityScoreList":[829,721,721],"projectPublicId":"6522662d-8795-40dd-afbf-28a2540a1c6b","projectUrl":"https://app.snyk.io/org/benjaminhuo/project/6522662d-8795-40dd-afbf-28a2540a1c6b?utm_source=github&utm_medium=referral&page=fix-pr","prType":"fix","templateFieldSources":{"branchName":"default","commitMessage":"default","description":"default","title":"default"},"templateVariants":["updated-fix-title","priorityScore"],"type":"auto","upgrade":["SNYK-DEBIAN12-GLIBC-5927132","SNYK-DEBIAN12-GLIBC-5927132","SNYK-DEBIAN12-GLIBC-6210098","SNYK-DEBIAN12-GLIBC-6210098","SNYK-DEBIAN12-SYSTEMD-6277507"],"vulns":["SNYK-DEBIAN12-GLIBC-5927132","SNYK-DEBIAN12-GLIBC-6210098","SNYK-DEBIAN12-SYSTEMD-6277507"],"patch":[],"isBreakingChange":false,"remediationStrategy":"vuln"}'

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/OpenFunction/functions-framework-python-gcp?pullRequest=18) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/OpenFunction/functions-framework-python-gcp?pullRequest=18) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/OpenFunction/functions-framework-python-gcp?pullRequest=18) it.</sub>

snyk-bot committed:
fix: examples/openfunction_dapr_trigger/caller/Dockerfile to reduce vulnerabilities

The following vulnerabilities are fixed with an upgrade:
- https://snyk.io/vuln/SNYK-DEBIAN12-GLIBC-5927132
- https://snyk.io/vuln/SNYK-DEBIAN12-GLIBC-5927132
- https://snyk.io/vuln/SNYK-DEBIAN12-GLIBC-6210098
- https://snyk.io/vuln/SNYK-DEBIAN12-GLIBC-6210098
- https://snyk.io/vuln/SNYK-DEBIAN12-SYSTEMD-6277507

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/OpenFunction/functions-framework-python?pullRequest=18) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/OpenFunction/functions-framework-python?pullRequest=18) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/OpenFunction/functions-framework-python?pullRequest=18) it.</sub>
